### PR TITLE
go/mio: add an example of REST API server

### PR DESCRIPTION
**Pull request by kiwionly2**
Signed-off-by: kiwionly2 <gheewooi.ong@seagate.com>

# Problem Statement
- Problem statement

# Design
-  For Bug, Describe the fix here.
-  For Feature, Post the link for design

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [ ] Test Cases cover Happy Path, Non-Happy Path and Scalability
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

-----
[View rendered bindings/go/go_motr_api_server/README.md](https://github.com/kiwionly2/cortx-motr/blob/go_api_server/bindings/go/go_motr_api_server/README.md)

### CODE REVIEW

**Comment by rkothiya** [comment]
Can one of the admins verify this patch?

**Comment by madhavemuri** [comment]
@andriytk : Can you please review this PR ?

**Review by andriytk — CHANGES_REQUESTED** [review]
Description should be updated with some details at least.

Also, try to fix Codacy issues as much as possible.

File: bindings/go/go_motr_api_server/README.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+It included a REST API server and a http client.
+
+The API implemented as below:
+
+```
+/api/kv/put
+/api/kv/get
+/api/kv/delete
+/api/object/read
+/api/object/write
```
Comment (andriytk) [inline bindings/go/go_motr_api_server/README.md:15]: What is `/api/`?

Comment (kiwionly2) [reply bindings/go/go_motr_api_server/README.md:15]: This is just part of the path of the url, should I remove it so the url will be much more simple ?

Comment (andriytk) [reply bindings/go/go_motr_api_server/README.md:15]: Yes, please.

Comment (kiwionly2) [reply bindings/go/go_motr_api_server/README.md]: Sorry for late reply, I just back from vacation, I will remove the "/api" path accrodingly.

File: bindings/go/go_motr_api_server/README.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+It included a REST API server and a http client.
+
+The API implemented as below:
+
+```
+/kv/put
+/kv/get
+/kv/delete
+/object/read
+/object/write
```
Comment (andriytk) [inline bindings/go/go_motr_api_server/README.md:15]: Is the leading slash `/` really needed?

File: bindings/go/go_motr_api_server/README.md (hunk 1 above)
Comment (andriytk) [reply bindings/go/go_motr_api_server/README.md]: I think the leading `/` is not needed also.

Comment (kiwionly2) [reply bindings/go/go_motr_api_server/README.md]: `/` removed.

**Comment by shailesh-vaidya** [comment]
Closing as an obsolete

**Pull request closed without merging**